nemo/collections/tts/data/datalayers.py

```diff
@@ -515,3 +522,235 @@ def preprocess_linear_specs_dataset(valid_filelist, train_filelist, n_fft, hop_l
+        sample_rate: int,
+        max_duration: Optional[int] = None,
+        min_duration: Optional[int] = None,
+        ignore_file: Optional[str] = None,
+        max_utts: int = 0,
+        trim: bool = False,
+        load_supplementary_values = True,   # Val LJSpeech files missing some supp. data
+    ):
+        super().__init__()
+
+        # Retrieve mappings from file
+        with open(mappings_filepath, 'r') as f:
+            mappings = json.load(f)
+            self.word2phones = mappings['word2phones']
+            self.phone2idx = mappings['phone2idx']
+
+        # Load data from manifests
+        audio_files = []
+        total_duration = 0
+        if isinstance(manifest_filepath, str):
+            manifest_filepath = [manifest_filepath]
+        for manifest_file in manifest_filepath:
+            with open(expanduser(manifest_file), 'r') as f:
+                logging.info(f"Loading dataset from {manifest_file}.")
+                for line in f:
+                    item = json.loads(line)
+                    audio_files.append({"audio_filepath": item["audio_filepath"], "duration": item["duration"]})
+                    total_duration += item["duration"]
+
+        total_dataset_len = len(audio_files)
+        logging.info(f"Loaded dataset with {total_dataset_len} files totalling {total_duration/3600:.2f} hours.")
+        self.data = []
+        if load_supplementary_values:
+            dataitem = py_collections.namedtuple(
+                typename='AudioTextEntity', field_names='audio_file duration text_tokens pitches energies'
+            )
+        else:
+            dataitem = py_collections.namedtuple(
+                typename='AudioTextEntity', field_names='audio_file duration text_tokens'
+            )
+
+        if ignore_file:
+            logging.info(f"using {ignore_file} to prune dataset.")
+            with open(ignore_file, "rb") as f:
+                wavs_to_ignore = set(pickle.load(f))
+
+        pruned_duration = 0
+        pruned_items = 0
+        for item in audio_files:
+            audio_path = item['audio_filepath']
+            LJ_id = os.path.splitext(os.path.basename(audio_path))[0]
+
+            # Prune data according to min/max_duration & the ignore file
+            if (min_duration and item["duration"] < min_duration) or (
+                max_duration and item["duration"] > max_duration
+            ):
+                pruned_duration += item["duration"]
+                pruned_items += 1
+                continue
+            if ignore_file and (LJ_id in wavs_to_ignore):
+                pruned_items += 1
+                pruned_duration += item["duration"]
+                wavs_to_ignore.remove(LJ_id)
+                continue
+
+            # Else not pruned, load additional info
+
+            # Phoneme durations and text token indices from durations file
+            dur_path = audio_path.replace('/wavs/', '/phoneme_durations_realign/').replace('.wav', '.pt')
```

Comment:
`calculate_durs.py` uses `/phone_durations/`, let's standardize.

Reply:
Whoops, that's a holdover from when I was testing, correcting it now.